### PR TITLE
feat: [EXT-3053] implement app details methods

### PR DESCRIPTION
**Pull request by Jwhiles**
## Checklist (check all before merging)

- [ ] Both unit and integration tests are passing
- [ ] There are no breaking changes
- [ ] Changes are reflected in the documentation

When adding a new method:

- [ ] The new method is exported through the default and plain CMA client
- [ ] All new public types are exported from `./lib/export-types.ts`
- [ ] Added a unit test for the new method
- [ ] Added an integration test for the new method
- [ ] The new method is added to the documentation

### CODE REVIEW

File: lib/entities/app-metadata.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import { Except } from 'type-fest'
+import { BasicMetaSysProps, DefaultElements, MakeRequest, SysLink } from '../common-types'
+import enhanceWithMethods from '../enhance-with-methods'
+
+type AppMetadataSys = Except<BasicMetaSysProps, 'version' | 'id'> & {
+  appDefinition: SysLink
+  organization: SysLink
+}
+
+// TODO should these be exported?
```
Comment (Jwhiles) [inline lib/entities/app-metadata.ts:12]: ?

File: tsconfig.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
       "AppInstalation",
       "AppBundle",
       "AppSignedRequest",
-      "AppSigningSecret",
+      "AppDetails",
```
Comment (andipaetzold) [inline tsconfig.json:37]: maybe also reorder things alphabetical

```suggestion
      "AppSigningSecret",
      "AppDetails",
```

**Review by andipaetzold — CHANGES_REQUESTED** [review]

File: lib/create-organization-api.ts [hunk 3]
```diff
@@ -684,5 +686,80 @@ export default function createOrganizationApi(makeRequest: MakeRequest) {
         /* noop*/
       })
     },
+    /**
+     * Creates or updates an app signing secret
```
[the remark this answers is not in the document]
Comment (Jwhiles) [reply lib/create-organization-api.ts:690]: Thanks! 

I thought I got all of these.

File: lib/create-organization-api.ts [hunk 4]
```diff
@@ -696,7 +696,9 @@ export default function createOrganizationApi(makeRequest: MakeRequest) {
      * })
      *
      * client.getOrganization('<org_id>')
-     * .then((org) => org.upsertAppDetails('app_definition_id', { value: 'tsren3s1....wn1e' }))
+     * .then((org) => org.upsertAppDetails('app_definition_id',
```
Comment (andipaetzold) [inline lib/create-organization-api.ts:699]: It feels strange to me that this method is on org level and not part of the app definition api (https://github.com/contentful/contentful-management.js/blob/master/lib/create-app-definition-api.ts) but we already did the same for the signing secret 🤷

**Review by andipaetzold — APPROVED** [review]

**Review by loweisz — APPROVED** [review]
🚀

**Pull request merged**